src/drivers/gitlab.js

```diff
@@ -244,15 +244,38 @@ class Gitlab {
     body.append('title', title);
     body.append('description', description);
 
-    const { web_url: url } = await this.request({
+    const { web_url: url, iid } = await this.request({
       endpoint,
       method: 'POST',
       body
     });
 
+    if (autoMerge) {
+      await new Promise((resolve) => setTimeout(resolve, 20000));
```

Comment:
during my testing, I found that Gitlab will throw "Method not Allowed" if the auto-merge request comes directly after creation. I consider this a bug, and will open an issue with them.

Reply:
Would it be better to retry the request in a loop with a smaller delay or [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff)? E.g. “while merge returns 405: wait seconds”

Reply:
definitely! let's wait what Gitlab responds, maybe there's an easier solution: https://gitlab.com/gitlab-org/gitlab/-/issues/353984

Reply:
In any case we also have a sleep function in utils.js

Reply:
> let's wait what Gitlab responds

I do not expect this to be fixed any time soon. GL can have issues for years

Reply:
added exponential backoff: https://github.com/iterative/cml/pull/903/commits/45491b78832713beef4e0f3ffd33c3a30f84f183